aws/config.go

```diff
@@ -78,6 +78,17 @@ type Config struct {
 	// Disabling this feature is useful when you want to use local endpoints
 	// for testing that do not support the modeled host prefix pattern.
 	DisableEndpointHostPrefix bool
+
+	// ExternalConfig defines how additional configuration can be loaded by service
+	// clients.
+	ExternalConfig ExternalConfig
+}
+
+// ExternalConfig is an interface that encapsulates the behavior of loading
+// additional configuration from an external source.
+type ExternalConfig interface {
+	// ResolveConfig calls the provide function passing slice of configuration sources
+	ResolveConfig(func(configs []interface{}) error) error
 }
```

Comment:
Does it make sense to rename this field to be more generic to additional config, instead of specific to `External`?   Even though that is what we plan to use it for.